Logger.Tests/FileLoggerTests.cs

```diff
@@ -1,10 +1,79 @@
 ﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.IO;
 
 namespace Logger.Tests
 {
     [TestClass]
     public class FileLoggerTests
     {
-        
+        [TestMethod]
+        public void FileLogger_AssignGoodFilePath_PathMatches()
+        {
+            //Arrange
+            FileLogger? logger = new FileLogger("testFile");
+
+            //Act
+            string path = "";
+            if (logger != null && !string.IsNullOrEmpty(logger.FilePath))
+                path = logger.FilePath;
+
+            //Assert
+            Assert.AreEqual(path, "testFile");
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void FileLogger_AssignNullFilePath_ReturnsArgumentNullException()
+        {
+            //Arrange
+
+            //Act
+            _ = new FileLogger(null!);
+
+            //Assert
+        }
+
+        [TestMethod]
+        public void FileLogger_AssignsGoodFilePath_LogMatchesInput()
+        {
+            if (!File.Exists("testFile.txt"))
+                throw new FileNotFoundException();
+
+            string filePath = "testFile.txt";
+
+            //Arrange
+            LogFactory? logFactory = new LogFactory();
+            logFactory.ConfigureFileLogger(filePath);
+            FileLogger? fileLogger = (FileLogger?)logFactory.CreateLogger(nameof(FileLogger));
```

Comment:
When unit testing you want to test a single _unit_ of code at a time. Since we are testing the `FileLogger` class and not the `LogFactory` class you shouldn't have any logic from the `LogFactory` class in your unit test. Instead, directly instantiate the `FileLogger` by doing:
```suggestion
            FileLogger fileLogger = new FileLogger(filePath);
```
You will have separate unit tests to ensure `ConfigureFileLogger()` and `CreateLogger()` work as expected.